packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2676,7 +2949,20 @@ class ObjectsInRedisClient {
         }
     }
 
-    getObject(id, options, callback) {
+    // cb version with options
+    getObject(id: string, options: Record<string, any> | undefined, callback: ioBroker.GetObjectCallback): void;
+    // Promise version
+    getObject(
+        id: string,
+        options?: Record<string, any>
+    ): Promise<ioBroker.CallbackReturnTypeOf<ioBroker.GetObjectCallback>>;
+    // no options but cb
+    getObject(id: string, callback: ioBroker.ErrorCallback): void;
+    getObject(
+        id: string,
+        options?: any,
+        callback?: ioBroker.ErrorCallback
+    ): void | Promise<ioBroker.CallbackReturnTypeOf<ioBroker.GetObjectCallback>> {
```

Comment:
Not sure if my comment got lost. I'd base this off the `getForeignObject` types to have TS automatically infer the object type from its ID:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/06f38fe1967d69ed164552b55674bf2d5269e7a1/types/iobroker/index.d.ts#L647-L651